src-cpp-structure/projects/App/Core/SignalsManager.cpp

```diff
@@ -0,0 +1,96 @@
+#include "SignalsManager.h"
+
+#include "GlobalEvents.h"
+#include <QtConcurrent>
+
+#include "libstatus.h"
```

Comment:
In theory yes. But there are also some signals sent from the status-go which are handled in SignalsManager class in the `Status.App` and if we do that in `Status.Backend` we would need to make it Qt dependable in order to send signals to/notify the rest of the app. Maybe in future we can rethink about that and create some better way of handling this.

Reply:
From what I see `SignalsManager` uses only `SetSignalEventCallback` from status-go, why not to introduce and use `Backend::registerSignalCallback` instead? It would fix that cross-layer call that breaks the N-layer architecture.

Reply:
We can do that.